### PR TITLE
Add Costing to BSM2 Flowsheet

### PR DESCRIPTION
**Pull request by MarcusHolly**
## Summary/Motivation:
Adds costing into the conventional BSM2 flowsheet

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by MarcusHolly** [comment]
There will be failures until #1186 and #1196 are merged in.

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py [hunk 1]
```diff
@@ -371,6 +365,10 @@ def set_operating_conditions(m):
     m.fs.RADM.volume_vapor.fix(300)
     m.fs.RADM.liquid_outlet.temperature.fix(308.15)
 
+    # Dewatering Unit - fix either HRT or volume.
+
+    # Set specific energy consumption averaged for centrifuge
+    m.fs.unit.energy_electric_flow_vol_inlet[0] = 0.069 * pyunits.kWh/pyunits.m**3
```
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py]:
```suggestion
    m.fs.DU.energy_electric_flow_vol_inlet[0] = 0.069 * pyo.units.kWh/pyo.units.m**3
```
I'm assuming this is what you meant @adam-a-a?

Comment (adam-a-a) [reply watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py]: Yes- thanks for the catch.

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py [hunk 2]
```diff
@@ -491,5 +532,130 @@ def display_results(m):
...
+    )
+    print(
+        "electricity consumption AD",
+        pyo.value(m.fs.RADM.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.RADM.electricity_consumption[0]),
+    )
+    print(
+        "electricity consumption dewatering Unit",
+        pyo.value(m.fs.DU.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.R5.electricity_consumption[0]),
```
Comment (bknueven) [inline watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py]:
```suggestion
        pyo.units.get_units(m.fs.DU.electricity_consumption[0]),
```

File: watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py [hunk 3]
```diff
@@ -491,5 +532,130 @@ def display_results(m):
...
+    )
+    print(
+        "electricity consumption thickening Unit",
+        pyo.value(m.fs.DU.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.R5.electricity_consumption[0]),
```
Comment (bknueven) [inline watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2.py]:
```suggestion
        pyo.value(m.fs.TU.electricity_consumption[0]),
        pyo.units.get_units(m.fs.TU.electricity_consumption[0]),
```

**Review by bknueven — COMMENTED** [review]
I found a couple copy+paste errors.

**Review by adam-a-a — APPROVED** [review]
LGTM - we expect to refine in subsequent PR

**Review by bknueven — APPROVED** [review]

**Pull request merged**